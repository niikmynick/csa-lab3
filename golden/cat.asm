.code
    read
    write
    pop

    halt
