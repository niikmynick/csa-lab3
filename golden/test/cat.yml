in_source: |-
  .code
    read
    write
    halt
in_stdin: |-
  400:T
  600:h
  680:i
  750:s
  900: 
  1000:i
  1100:s
  1150: 
  1200:a
  1230: 
  1500:w
  1600:a
  2000:y
  2200:\0
out_bin: |-
  1111111111111110001000000000001100000000000000000000000000000000000000000000000000000000001000010000001100000000000000000000000000000000000000000000000000000000010000000000001100000000000000000000000000000000000000000000000000000000
out_stdout: |-
  This is a way
out_log: |-
  INFO      ::     INPUT       Input schedule: [(400, 'T'), (600, 'h'), (680, 'i'), (750, 's'), (900, ' '), (1000, 'i'), (1100, 's'), (1150, ' '), (1200, 'a'), (1230, ' '), (1500, 'w'), (1600, 'a'), (2000, 'y'), (2200, '\\0')]
  INFO      ::     SYSTEM      Starting execution
  DEBUG     ::     SYSTEM      Initial state: ControlUnit: time=0, instructions executed=0
  DEBUG     ::     SYSTEM      Instructions memory: {0: READ - 203, 1: WRITE - 213, 2: HALT - 403}
  DEBUG     ::     SYSTEM      Data memory: {0: InputManager, 1: OutputManager}
  DEBUG     ::     INTER       Have 0 interruptions
  DEBUG     ::     SYSTEM      ControlUnit: time=5, instructions executed=0
  INFO      ::     INSTR       Processing READ
  DEBUG     ::     INTER       Have 1 interruptions
  DEBUG     ::     INTER       Processing Interruption: INPUT
  INFO      ::     INPUT       Reading input
  INFO      ::     INPUT       Read value: This is a way
  DEBUG     ::     INTER       Finished processing Interruption: INPUT
  DEBUG     ::     SYSTEM      ControlUnit: time=2208, instructions executed=1
  INFO      ::     INSTR       Processing WRITE
  DEBUG     ::     INTER       Have 1 interruptions
  DEBUG     ::     INTER       Processing Interruption: OUTPUT
  INFO      ::     OUTPUT      Wrote This is a way
  DEBUG     ::     INTER       Finished processing Interruption: OUTPUT
  DEBUG     ::     SYSTEM      ControlUnit: time=2220, instructions executed=2
  INFO      ::     INSTR       Processing HALT
  DEBUG     ::     INTER       Have 1 interruptions
  DEBUG     ::     INTER       Processing Interruption: HALT
  INFO      ::     SYSTEM      Halting
  INFO      ::     SYSTEM      Execution finished, state: ControlUnit: time=2227, instructions executed=3
  INFO      ::     INPUT       Turning off input device.
  INFO      ::     OUTPUT      Turning off output device.
  
